### PR TITLE
improve detection whether is Java GraalVM

### LINKED ISSUE #1351: Improve detection of running in graalvm
https://github.com/graalvm/polyglot-embedding-demo/blob/main/build.gradle.kts#L30

### PR DESCRIPTION
**Pull request by sdelamo**
Close: #1351
see: https://github.com/graalvm/polyglot-embedding-demo/blob/main/build.gradle.kts#L30

### CODE REVIEW

File: buildSrc/src/main/groovy/io/micronaut/guides/GuidesPlugin.groovy [hunk 1]
```diff
@@ -434,11 +436,12 @@ class GuidesPlugin implements Plugin<Project> {
...
-        return Arrays.asList("jvmci.Compiler", "java.vendor.version", "java.vendor")
+        (System.getProperty("java.home") != null && Files.exists(Paths.get("${System.getProperty("java.home")}/lib/graalvm")))
+                || Arrays.asList("jvmci.Compiler", "java.vendor.version", "java.vendor")
                 .stream()
                 .anyMatch(propertyName -> {
                     String value = System.getProperty(propertyName)
-                    return value != null && value.toLowerCase(Locale.ENGLISH).contains("graal");
+                    value != null && value.toLowerCase(Locale.ENGLISH).contains("graal")
                 })
     }
```
Comment (timyates) [inline buildSrc/src/main/groovy/io/micronaut/guides/GuidesPlugin.groovy]:
```suggestion
    private static boolean isGraalVMJava() {
        (System.getProperty("java.home") != null && Files.exists(Paths.get("${System.getProperty("java.home")}/lib/graalvm"))) ||
                ["jvmci.Compiler", "java.vendor.version", "java.vendor"].any {
                    System.getProperty(it)?.toLowerCase(Locale.ENGLISH)?.contains("graal")
                }
    }
```

**Review by timyates — APPROVED** [review]

File: buildSrc/src/main/groovy/io/micronaut/guides/GuidesPlugin.groovy (hunk 1 above)
Comment (sdelamo) [reply buildSrc/src/main/groovy/io/micronaut/guides/GuidesPlugin.groovy]: resolved via https://github.com/micronaut-projects/micronaut-guides/pull/1354/commits/891fac444a6c741a4268b192330ecbae118e5e1a

**Review by timyates — APPROVED** [review]
Maybe we should just check the path exists?

Tested locally with:

```
ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
```

And with a GraalJDK I get:

```
❯ ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
jvmti-agent-base.jar         launcher-common.jar          svm-agent.jar                svm-configure.jar            svm-diagnostics-agent.jar    svm-driver.jar               svm-enterprise-configure.jar
```

And with zulu I get

```
❯ ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
ls: /Users/tim/.sdkman/candidates/java/17.0.7-zulu/zulu-17.jdk/Contents/Home/lib/graalvm: No such file or directory
```

**Comment by sdelamo** [comment]
> Maybe we should just check the path exists?
> 
> Tested locally with:
> 
> ```
> ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
> ```
> 
> And with a GraalJDK I get:
> 
> ```
> ❯ ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
> jvmti-agent-base.jar         launcher-common.jar          svm-agent.jar                svm-configure.jar            svm-diagnostics-agent.jar    svm-driver.jar               svm-enterprise-configure.jar
> ```
> 
> And with zulu I get
> 
> ```
> ❯ ls $(groovy -e "println System.properties.'java.home'")/lib/graalvm
> ls: /Users/tim/.sdkman/candidates/java/17.0.7-zulu/zulu-17.jdk/Contents/Home/lib/graalvm: No such file or directory
> ```

what are you suggesting exactly?

**Comment by timyates** [comment]
Sorry I was suggesting just to have

```
    sys("java.home") != null && Files.exists(Paths.get("${sys("java.home")}/lib/graalvm"))
```

The same as they have in https://github.com/graalvm/polyglot-embedding-demo/blob/main/build.gradle.kts#L30

But it's fine as it is 👍

**Pull request merged**